source/include/ota.h

```diff
@@ -455,6 +654,34 @@ void OTA_EventProcessingTask( void * pUnused );
  *
  * @param[in] pEventMsg Event to be added to the queue
  * @return true If operation is successful, false If the event can not be added
+ *
+ * <b>Example</b>
+ * Signal OTA agent that a new file block has been received over the http connection.
+ * @code{c}
+ * OtaHttpStatus_t handleDataFromHTTPService( const HTTPResponse_t * pResponse )
+ * {
+ *     // Assume otaEventBufferGet defined to get synchronized event buffers.
+ *     OtaEventData_t * pData = otaEventBufferGet();
+ *     bool result = false;
+ *
+ *     if( pData != NULL )
+ *     {
+ *         memcpy( pData->data, pResponse->pBody, pResponse->bodyLen );
```

Comment:
Not validating `pData->data`, `pResponse`, or `pResponse->pBody` here – if developers need to do those validations then they should be added.

Reply:
Added a comment: https://github.com/divekarshubham/ota-for-aws-iot-embedded-sdk-1/blob/25a6f8f1ad0fe39121247eb17488209ad6c5135d/source/include/ota.h#L685